### ISSUE TITLE
Error: nested exception

### ISSUE THREAD

**Issue by Romounet** [GoSecure/pyrdp#79]
An excception seem to occur inside an exception handler (don't know how to reproduce this):

```Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/core/packing.py", line 26, in unpack
    return struct.unpack(cls.FORMAT, data.read(length))[0]
struct.error: unpack requires a buffer of 2 bytes

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/layer/tcp.py", line 81, in dataReceived
    self.next.recv(data)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/layer/segmentation.py", line 88, in recv
    layer.recv(forwarded)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/layer/buffered.py", line 52, in recv
    self.pduReceived(pdu, self.hasNext)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/layer/layer.py", line 126, in pduReceived
    self.next.recv(pdu.payload)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/layer/x224.py", line 79, in recv
    self.pduReceived(pdu, pdu.header == X224PDUType.X224_TPDU_DATA)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/layer/layer.py", line 126, in pduReceived
    self.next.recv(pdu.payload)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/layer/mcs.py", line 29, in recv
    self.pduReceived(pdu, self.hasNext)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/layer/layer.py", line 123, in pduReceived
    self.observer.onPDUReceived(pdu)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/core/observer.py", line 84, in __call__
    self.composite.doCall(self.item, args, kwargs)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/core/observer.py", line 56, in doCall
    getattr(observer, item)(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/layer/layer.py", line 48, in onPDUReceived
    self.handlers[pdu.header](pdu)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/mcs/server.py", line 152, in onSendDataRequest
    user.recvSendDataRequest(pdu.channelID, pdu)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/mcs/user.py", line 91, in recvSendDataRequest
    self.channels[channelID].recvSendDataRequest(data)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/mcs/channel.py", line 99, in recvSendDataRequest
    self.pduReceived(pdu, True)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/layer/layer.py", line 126, in pduReceived
    self.next.recv(pdu.payload)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/layer/rdp/security.py", line 151, in recv
    self.next.recv(data)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/layer/rdp/slowpath.py", line 90, in recv
    pdu = self.mainParser.parse(data)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/parser/rdp/slowpath.py", line 65, in parse
    return self.parsers[header.pduType](stream, header)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/parser/rdp/slowpath.py", line 73, in parseData
    return self.dataParsers[header.subtype](stream, header)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/parser/rdp/slowpath.py", line 468, in parseSuppressOutput
    left = Uint16LE.unpack(stream)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/core/packing.py", line 28, in unpack
    raise ValueError(str(e))
ValueError: unpack requires a buffer of 2 bytes
[2019-02-27 13:53:14,320] - mitm.connections.Orlando949591.server              - INFO     - Connection closed: [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionLost'>: Connection to the other side was lost in a non-clean fashion: Connection lost.
]```

**Comment by Res260** [comment]
Do you know what kind of machine connected to the MITM?

**Comment by Romounet** [comment]
Client was a windows 10.
Sadly server is not logged, so I can't say for sure but I think it was another Windows 10 machine

**Comment by xshill** [comment]
Looks like the same bug as #77

**Comment by xshill** [comment]
Fixed in a5fab9b171df4091370e585086c432e710eb4834
